ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity.component.ts

```diff
@@ -354,6 +322,47 @@ export class ActivityComponent extends BaseActivityComponent implements OnInit,
         return -1;
     }
 
+    // Navigate through fields
+    protected nextField(event: Event, {elements, preventDefault = false, carousel = false}: NavigateFieldsOptionsModel): void {
+        preventDefault && event.preventDefault();
+        if(elements && elements.length > 0) {
+            // All provided elements array, excluding those, which doesn't have id or style.display is 'none'
+            const elementsArray: Element[] = Array.from(document.querySelectorAll(elements.join(',')))
+                .filter(input => input.id && (input as HTMLElement)?.style.display !== 'none');
+            // Map data structure for elements in order to easily find them
+            const elementsObjects: Map<Element, number> = new Map(elementsArray.map((input, i) => [input, i]));
+            // Pressed element in a DOM
+            const pressedElement = event.target as Element;
+            // One-sided or Carousel behaviour while moving through fields
+            this.switchBehaviour(elementsArray, elementsObjects.get(pressedElement), carousel);
+        }
+    }
+    // Decide behaviour while navigation
+    private switchBehaviour(elements: Element[], elementIndex: number | undefined, carousel: boolean): void {
+        if(carousel){
+            if(elements.length - 1 === elementIndex) {
+                this.carouselRevert = true;
+            }
+            if(elementIndex === 0) {
+                this.carouselRevert = false;
+            }
```

Comment:
Just wondering whether it is enough to check for the last index only? 
Like: 
```
 if(elements.length - 1 === elementIndex) {
                this.carouselRevert = true;
            }
            if(elementIndex === 0) {
                this.carouselRevert = false;
            }
```
=> `this.carouselRevert =  elementIndex === elements.length - 1;`

Reply:
Do you have in mind any case, where it would fail?

Reply:
https://github.com/broadinstitute/ddp-angular/pull/1092#discussion_r822547606
Can you please check the case ?